content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js

```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
+			$("#" + question + "-progress-label").addClass("progress-hidden");
+			$("#" + question + "-progress-label").removeClass("progress");
+
+			switch(question) {  // here's where flow is really controlled
+				case name+"-array-dec":		
+					next = name+"-array-ref";	
+					break;
+				case name+"-array-ref":	
+					next = name+"-var-range";	
+					break;
+				case name+"-var-range":		
+					next = name+"-var-occur";	
+					break;
+				case name+"-var-occur":	
+					next = name+"-var-modify";	
+					break;
+				case name+"-var-modify":	
+					next = name+"-var-array";	
+					break;
+				
+					default:	
+					return;	// input invalid or question is complete
+			}
+		}
+
+		// focus on next question
+		$("#"+next+"-label").addClass("si-checklist-active");
+		// if the question only has 1 span to click, don't show progress bar
+		if(next.endsWith("-array-dec") || next.endsWith("-array-ref") || next.endsWith("-var-array")){ }else{
+			// remove hidden class from progress bar
+			$("#" + next + "-progress-label").removeClass("progress-hidden");
+			// add the progress bar fill
+			$("#" + next + "-progress-label").addClass("progress");
+		}
+
+		//IF POPOVER IS NEEDED
+		if(next === name+"-var-range") { 
+			
+			// If popover is needed, remove progress bar
+			if (
+			next.endsWith("var-occur") || 
+			next.endsWith("var-range")
+			) {  // remove progress bar
+				$("#"+next+"-progress-label").removeClass("progress");
+				$("#"+next+"-progress-label").addClass("progress-hidden");
+			}
+			
+			waitingOn = [];  // gather spans remaining to be clicked
+			
+			$(".span-"+name).each(function(index) {
+				if( $(this).hasClass(next+"-"+name) ) {
+					waitingOn.push(index);
 				}
-
-				else{ //normal question
-
-					waitingOn = [];
-					$(".span-"+name).each(function(index) {
-						if( $(this).hasClass(next+"-"+name) ) {
-							waitingOn.push(index);
-						}
-					});
+			});
+
+			showPopOver(spans[7]); // show Popover over the index span
+			var tag = "Pgm1CL-var-range"; //the one that needs the popover
+			
+			$('input[name=popSave2]').on('click', function() {
+				if ($('#lowInt').val() == "0" && $('#popSelect1').val() == "<="
+					&& $('#popSelect2').val() == "<" && $('#highInt').val() == "10"
+				) { // these need to change to the right answer for the question
+					$('.rangepop').hide();
+					$(".span-"+name).click(); //click the empty span so that next question is asked.
+					$('.popover').css('background-color', 'white');
+				} else {
+					$('.popover').css('background-color', 'lightpink');
 				}
+			});
+		} else { //NORMAL QUESTION
 
-					return next;
+			waitingOn = [];
+			$(".span-"+name).each(function(index) {
+				if( $(this).hasClass(next+"-"+name) ) {
+					waitingOn.push(index);
 				}
+			});
+		}	
+		//removes highlighted sections so user is not seeing correct answers of previous question
+		$(".span-" + name).removeClass("si-code-clicked");
+		return next;
```

Comment:
This is where this code block should appear if the checklist only has multi-span questions. At the moment this statement is shown in both spots. If the checklist has single-span questions, you would delete this and keep the one that's at the top of this function